DirectEmployers/MyJobs
# [PD-1839] Apply reusable table component

# Overview

This PR covers the application of a React table component that can be used in various applications.

The application is on the Non User Outreach project as a list of various Non User Outreach records. Application.

## Setup

To test this table, you first need to run migrations on mypartners

Afterward, you will need to execute the following script to create the outreach records needed. NOTE: You do not need to use Facebook as the company. Change that bit with any company you currently are a user/admin of.

```
from datetime import datetime
from factory import fuzzy
from seo.models import Company
from mypartners.models import (OutreachRecord, OutreachEmailAddress,
                               OutreachWorkflowState)

company = Company.objects.get(name="Facebook")
try:
    outreach_email = OutreachEmailAddress.objects.get(company=company,
                                                      email="testemail")
except:
    outreach_email = OutreachEmailAddress(company=company,
                                          email="testemail")
    outreach_email.save()

try:
    workflow = OutreachWorkflowState.objects.get(state="Reviewed")
except:
    workflow = OutreachWorkflowState(state="Reviewed")
    workflow.save()

i = 0
while i < 100:
    i += 1
    OutreachRecord(date_added=datetime.now(),
                   outreach_email=outreach_email,
                   from_email=str(fuzzy.FuzzyText(length=10).fuzz()) + "@example.com",
                   email_body=fuzzy.FuzzyText(length=200).fuzz(),
                   current_workflow_state=workflow
                   ).save()

```

## Testing

* Run npm install
* Run gulp/dev server build
* Run secure locally
* Navigate to https://secure.my.jobs/prm/view/nonuseroutreach (or whatever url you use to access your secure instance)
* Click Outreach Records
* Verify table works properly.. Pagination is turned off for IE8 as there is an issue with the module we are using.
